05_concurrency/homework/main.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+type Cache interface {
+	Set(key string, value interface{}, ttl time.Duration)
+	Get(key string) (interface{}, bool)
+	Delete(key string)
+}
+
+type Item struct {
+	value       interface{}
+	timeCreated time.Time
+	ttl         time.Duration
+}
+
+type memoryCache struct {
```

Comment:
It should either have constructor or check whether map is nil on `Set` or have exported map field in order to allow others use this type by importing as a package. Of course, package name will not be `main` this way.